_posts/news/2017-03-21-announcing-scalajs-0.6.15.md

```diff
@@ -0,0 +1,372 @@
+---
+layout: post
+title: Announcing Scala.js 0.6.15
+category: news
+tags: [releases]
+permalink: /news/2017/03/21/announcing-scalajs-0.6.15/
+---
+
+
+We are excited to announce the release of Scala.js 0.6.15!
+
+This release was focused on preparations for Scala.js 1.0.0.
+In particular, it should contain the last deprecations for things that are being replaced by better alternatives.
+If your code and build compile without warnings with 0.6.15, it should continue to work unchanged in 1.0.0.
+
+The new deprecations come with a set of exciting new features, including several awesome JavaScript interoperability features:
+
+* Use `@JSExportStatic` to define static members on Scala.js-defined JS classes.
+* `@JSExportTopLevel` is not restricted to methods anymore: it can be used on fields and top-level classes/objects.
+* The standard library now contains the API for ECMAScript 2015 symbols (as [`js.Symbol`]({{ site.production_url }}/api/scalajs-library/latest/#scala.scalajs.js.Symbol)) and iterables/iterators (as [`js.Iterable`]({{ site.production_url }}/api/scalajs-library/latest/#scala.scalajs.js.Iterable) and [`js.Iterator`]({{ site.production_url }}/api/scalajs-library/latest/#scala.scalajs.js.Iterator)).
+* In JS types (both native and non-native), `@JSName` can be given a reference to a `js.Symbol` (instead of a constant string) to declare members whose "name" is a JavaScript symbol.
+
+Finally, the sbt plugin also features a replacement for the "launcher" files, so that the call to the main method of your program can be included directly inside the main `-fastopt.js`.
+
+Read on for more details.
+
+<!--more-->
+
+## Getting started
+
+If you are new to Scala.js, head over to
+[the tutorial]({{ BASE_PATH }}/tutorial/).
+
+## Release notes
+
+If upgrading from Scala.js 0.6.12 or earlier, make sure to read [the release notes of 0.6.13]({{ BASE_PATH }}/news/2016/10/17/announcing-scalajs-0.6.13/), which contain some breaking changes in sbt build definitions.
+
+As a minor release, 0.6.15 is backward source and binary compatible with previous releases in the 0.6.x series.
+Libraries compiled with earlier versions can be used with 0.6.15 without change.
+However, it is not forward compatible: libraries compiled with 0.6.15 cannot be used by projects using 0.6.{0-14}.
+
+Please report any issues [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Main deprecations
+
+### `@js.native` classes and objects need to have `@JSGlobal`
+
+Previously, one would declare facade types for native classes and objects as follows:
+
+{% highlight scala %}
+@js.native
+object Foo extends js.Object
+
+@js.native
+@JSName("Foobar")
+class Bar extends js.Object
+{% endhighlight %}
+
+This is now deprecated: `@js.native` objects and classes should now explicitly use the annotation `@JSGlobal` (as opposed to `@JSImport`) as follows:
```

Comment:
Hum ... I really meant `JSImport`, in the sense that either you have `@JSGlobal` or `@JSImport`. I guess the parentheses are confusing. I'll remove that.